src/types/common.ts

```diff
@@ -88,3 +88,22 @@ export interface AoClient {
   unmonitor: typeof unmonitor;
   dryrun: typeof dryrun;
 }
+
+export interface AOContract {
```

Comment:
in the constructor we could pass in a response parser - for example, an ANTResponseParser class or function that has a map for validating happy and sad paths.

Would prevent having to specify parsing of schemas in each method on implementations

Reply:
I refactored - but did not use the data attribute. In this particular case I think the JSX approach is cleaner.

If I had a non binary value that would require a multiline JSX case then I would probably use the data attribute though. Pretty cool that tailwind allows that.

Reply:
see comment above